### PR TITLE
ASoC: SOF: topology: remove the duplicate 'core' argument

### PR DESCRIPTION
**Pull request by keyonjie**
Since we can get the DSP core index from swidget->core, it is duplicated
to pass the extra 'core' argument for those sof_widget_load_xx()
functions.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>

### CODE REVIEW

**Review by dbaluta — DISMISSED** [review]

**Review by paulstelian97 — DISMISSED** [review]

**Review by kv2019i — DISMISSED** [review]

**Comment by keyonjie** [comment]
Thanks fore approval @kv2019i  @paulstelian97  and @dbaluta. Just pushed a minor update for the travis-ci complaining about excess description remained for the 'core' argument, can you help to re-grant your approvals?
```
sound/soc/sof/topology.c:1474: warning: Excess function parameter 'core' description in 'sof_comp_alloc'
```

**Review by paulstelian97 — APPROVED** [review]

**Comment by plbossart** [comment]
@keyonjie this work essentially undoes all the changes suggested by @lyakh that we just upstreamed.

I feel there's too much churn here, we can't just change the code back and forth this way.

**Comment by keyonjie** [comment]
> @keyonjie this work essentially undoes all the changes suggested by @lyakh that we just upstreamed.
> 
> I feel there's too much churn here, we can't just change the code back and forth this way.

@plbossart got your concern, but cleaning up and simplification should be welcome always no?

**Comment by keyonjie** [comment]
@lyakh Guennadi need your help to review and comment if the removal here is safe enough.

File: sound/soc/sof/topology.c [hunk 1]
```diff
@@ -2450,7 +2448,7 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 			return -ENOMEM;
 		}
 
-		ret = sof_widget_load_dai(scomp, index, swidget, comp.core,
+		ret = sof_widget_load_dai(scomp, index, swidget,
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: nitpick: some of these lines could be merged into 1

**Review by lyakh — APPROVED** [review]
Thanks for the patch, is indeed better this way!

File: sound/soc/sof/topology.c (hunk 1 above)
Comment (keyonjie) [reply sound/soc/sof/topology.c]: > nitpick: some of these lines could be merged into 1

yes, I merged some of them but with some unchanged, I was hesitate about if I should follow the 80 characters or the new 100 rule when I was making the commit.

Let me follow the 100-character maximum rule and merge all of them if possible.

File: sound/soc/sof/topology.c [hunk 2]
```diff
@@ -2461,11 +2458,10 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
...
 		break;
 	case snd_soc_dapm_mixer:
-		ret = sof_widget_load_mixer(scomp, index, swidget, comp.core,
-					    tw, &reply);
+		ret = sof_widget_load_mixer(scomp, index, swidget, tw, &reply);
 		break;
 	case snd_soc_dapm_pga:
-		ret = sof_widget_load_pga(scomp, index, swidget, comp.core,
+		ret = sof_widget_load_pga(scomp, index, swidget,
 					  tw, &reply);
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: yep, almost there ;-)

**Review by lyakh — CHANGES_REQUESTED** [review]

File: sound/soc/sof/topology.c (hunk 2 above)
Comment (keyonjie) [reply sound/soc/sof/topology.c]: eagle eyes!

**Review by lyakh — APPROVED** [review]
thanks!

**Review by kv2019i — APPROVED** [review]
Thanks, looks good! Indeed it's a bit unfortunate I just sent the multi-core patchset to upstream and it got merged ,and we need to patch it again, but alas, I do think this is better. I think I'll wait a bit if some more last-minute changes come, and we can send an update early next week (the latest).

File: sound/soc/sof/topology.c [hunk 3]
```diff
@@ -2229,7 +2227,7 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	}
 
 	process = (struct sof_ipc_comp_process *)
-		  sof_comp_alloc(swidget, &ipc_size, index, 0);
```
Comment (plbossart) [inline sound/soc/sof/topology.c:2232]: this is actually a functional change, your commit message does not capture that. It's not just a replacement, it fixes an error that resulted in the process being scheduled on core0. You also need a Fixes tag for this, it'll make the upstream contribution easier.

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/sof/topology.c (hunk 3 above)
Comment (keyonjie) [reply sound/soc/sof/topology.c:2232]: good point @plbossart let me change it.

**Review by juimonen — APPROVED** [review]
lgtm, all experts in the field seem to agree and comments are addressed.

**Review by lyakh — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Good point from Pierre. This looks good now.

**Review by plbossart — APPROVED** [review]
thanks @keyonjie

**Pull request merged**